Framework/API/src/FileProperty.cpp

```diff
@@ -396,5 +401,55 @@ std::string FileProperty::convertExtension(const std::string &filepath) const {
   fullpath.setExtension(ext);
   return fullpath.toString();
 }
+
+/** Expand user variables in file path.
+ *  On UNIX, ~ is replaced by the user's home directory, if found.
+ *  On Windows, a combination of HOMEPATH and HOMEDRIVE will be used.
+ *  If the path contains no user variables, or expansion fails, the path is
+ *  returned unchanged, for errors to be dealt with by the calling function
+ *  @param filepath The path to expand
+ *  @return The expanded path
+ */
+std::string FileProperty::expandUser(const std::string &filepath) const {
+  auto start = filepath.begin();
+
+  if (*start != '~') // No user variables in filepath
+    return filepath;
+
+  // Position of the first slash after the variable
+  auto nextSlash = find_if(start, filepath.end(), &isSlash);
+
+  // UNIX-style home variable (ie ~/blah)
+  if (std::distance(start, nextSlash) == 1) {
+    char *home;
+
+    if (!(home = std::getenv("HOME")))
+      if (!(home = std::getenv("USERPROFILE")))
+        if (!(home = std::getenv("HOMEPATH")))
```

Comment:
I think it works on my Windows 7. However I'm inclined to agree that we should just check for `~` - copying the behaviour from `ntpath` and `posixpath` has the potential to give us some real headaches. The library that makes expanding `~user/blah` feasible on UNIX doesn't exist on Windows, so I'm currently running into some really nasty (possibly really error-prone) workarounds with `ifdef`. Does anyone actually use the `~user/blah` syntax?